spec/features/recipes/show_spec.rb

```diff
@@ -0,0 +1,40 @@
+require "rails_helper"
+
+RSpec.describe "recipes show page", type: :feature do
+  it "shows a recipe by id number with its attributes and a list of ingredients" do
+    spaghet = Recipe.create!(name: "Spaghetti", complexity: 2, genre: "Pasta")
+    cheese = spaghet.ingredients.create!(name: "Cheese", cost: 3)
+    noods = spaghet.ingredients.create!(name: "Noodles", cost: 2)
+    herbals = spaghet.ingredients.create!(name: "Herbs", cost: 1)
+    sauce = spaghet.ingredients.create!(name: "Red Sauce", cost: 5)
+
+    visit "/recipes/#{spaghet.id}"
+
+    expect(current_path).to eq("/recipes/#{spaghet.id}")
+    expect(page).to have_content(spaghet.name)
+    expect(page).to have_content(spaghet.complexity)
```

Comment:
These assertions are good, but could be more robust if you anchored them to text they should appear next to. You've set up your view well with labels next to each of these attributes, so you could change your assertions to `    expect(page).to have_content("Complexity: #{spaghet.complexity}")`

This will make sure that this data is located in the correct place, and is meaningfully rendered on the page.